Expand setup instructions and update/fix npm dependencies

kfranqueiro opened the pull request:
I ran into trouble getting this set up under Debian and it took a while to figure out the resolution, so I figured I should probably document what I ran into in case anyone else runs into it.

Upon further investigation, I found that it has become impossible to run `npm i` with the current dependencies in this repo, and indeed it seems this had been worked around for Sass purposes by reverting to using Jekyll's process (#264), which loses the remaining steps in the gulp task (autoprefixer, px-to-rem, minification).

I have resolved the issue with dependencies by updating `gulp-sass` and installing `sass` (dart-sass), reverted most of #264 (leaving the `_config.yml` change intact), and have verified that the output CSS is the same with the exception of the way colors are encoded and some vendor prefixes that are no longer necessary.

kfranqueiro committed:
README: Expand setup instructions and document Linux gotcha

kfranqueiro commented:
@iadawn I attempted to also update the part of the README mentioning `npm i` for working on CSS/JS, but I am realizing that #264 almost certainly makes the gulp CSS workflow no longer work... given that there have seemingly been CSS changes since then, how do you build this now? (e.g. I imagine Jekyll handles the CSS entirely now, but do we still make use of the `js` gulp task? And was #264 fully intentional or a workaround for the sass-related dependencies causing install issues? I imagine Jekyll won't give us autoprefixer / px-to-rem / minification...)

iadawn commented:
Manually mostly. I could never get the gulp build to work and the changes I was making were small so it was quicker just to hack them in.

kfranqueiro committed:
Update gulp-sass; update package-lock; remove outdated pnpm lock

kfranqueiro committed:
Mostly revert "Use jekyll scss (#264)"

This reverts commit 53032c4dea31b2a7280a1a3b20c1001b88033dba,
except for the _config.yml changes.

kfranqueiro committed:
Add and document npm scripts

kfranqueiro committed:
Update compiled CSS assets; silence dart-sass deprecation warnings

kfranqueiro committed:
Rebuild styles with updated caniuse-lite

kfranqueiro committed:
Update dependencies within same major versions

kfranqueiro commented:
> Manually mostly. I could never get the gulp build to work and the changes I was making were small so it was quicker just to hack them in.

Indeed, the npm dependencies were severely out of date which prevented even installing dependencies. I have resolved that and reverted to the existing gulp task for building CSS (and added documentation on how it can be run).
